src/v/raft/state_machine.cc

```diff
@@ -123,7 +123,12 @@ ss::future<> state_machine::apply() {
       .then([this] {
           auto f = ss::now();
           if (_next < _raft->start_offset()) {
-              f = handle_eviction();
+              f = handle_eviction().then([this] {
+                  // if after the handle_eviction() call _next advances, this is
+                  // equivalent to the state machine applying all records up to
+                  // (not including) _next, so we have to notify waiters.
+                  _waiters.notify(model::prev_offset(_next));
```

Comment:
i think we do not need that as it is already done as a part of set next

Reply:
yeah I fixed it independently, but now it is not needed